crypto/stateproof/weights_test.go

```diff
@@ -184,6 +186,14 @@ func TestNumReveals(t *testing.T) {
 	}
 }
 
+func TestSigPartProofMaxSize(t *testing.T) {
+	partitiontest.PartitionTest(t)
+	t.Parallel()
+	// Ensures that the SigPartProofMaxSize constant used for totalallocbound for StateProof.(Sig|Part)Proof(s) is
+	// correct. It should be logically bound by the maximum number of StateProofTopVoters. It is scaled by 1/2
```

Comment:
Why is it scaled by 1/2?

Reply:
This comes from Idan's explanation and you perhaps understand it better but my understanding is that this is the worst case scenario but even so you only need one side of the tree for each of the voters.

Reply:
It is from my explanation [here](https://github.com/algorand/go-algorand/pull/3171/files)  :-D
It is not one side of the tree. It is more complicated than that. 

The hints are the values needed to compute the root given the reveals (leaves with `^` under them in the illustration below).
The idea is the following: 

```
	// Path is bounded by MaxNumLeaves since there could be multiple reveals, and
	// given the distribution of the elt positions and the depth of the tree,
	// the path length can increase up to 2^MaxTreeDepth / 2

Consider two different reveals for the same tree:
               z5
       z3              z4
   y       z       z1      z2
 q   r   s   t   u   v   w   x
a b c d e f g h i j k l m n o p
  ^
  hints: [a, r, z, z4]
len(hints) = 4
You need a to combine with b to get q, need r to combine with the computed q and get y, and so on.

The worst case is this:
               z5
       z3              z4
   y       z       z1      z2
 q   r   s   t   u   v   w   x
a b c d e f g h i j k l m n o p
^   ^     ^   ^ ^   ^     ^   ^

hints: [b, d, e, g, j, l, m, o]
len(hints) = 2^4/2
```